From b874548ef508b10bbc8836c44d6dfc933d420090 Mon Sep 17 00:00:00 2001
From: Rekha Eswaran <rekha.eswaran@intel.com>
Date: Fri, 6 Sep 2019 17:15:47 +0800
Subject: [PATCH] DRV_DPM_SW-32 - To fix the observed linux kernel cpufreq
 driver and DP scheduling caused when DP COC module init is called with
 CPUFREQ enabled for xrx500 power cycle test

---
 drivers/net/datapath/dpm/gswip30/datapath_coc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/datapath/dpm/gswip30/datapath_coc.c b/drivers/net/datapath/dpm/gswip30/datapath_coc.c
index 9648fbf4f743..35dfcdac9c18 100644
--- a/drivers/net/datapath/dpm/gswip30/datapath_coc.c
+++ b/drivers/net/datapath/dpm/gswip30/datapath_coc.c
@@ -755,7 +755,6 @@ int dp_coc_cpufreq_init(void)
 	PR_INFO("enter dp_coc_cpufreq_init\n");
 	spin_lock_init(&dp_coc_lock);
 	gsw_handle = dp_port_prop[inst].ops[GSWIP_R];
-	cpufreq_update_policy(0);
 
 	if (!polling_period)
 		polling_period = 2;
@@ -774,6 +773,7 @@ int dp_coc_cpufreq_init(void)
 	dp_coc_timer.function = dp_rmon_polling;
 	dp_coc_init_stat = 1;
 	dp_coc_ena = 1;
+	schedule_work(&coc_work_q);
 	PR_INFO("Register DP to CPUFREQ successfully.\n");
 	return 0;
 }
